### DISCUSSION TITLE
TextMeasurer.TryMeasureCharacterBounds doesn't return bounds for some characters using Style Script font

### DISCUSSION THREAD

**Discussion by Kizuto3** [SixLabors/Fonts#401] [Q&A] [answered]
I ran into a problem where TextMeasurer.TryMeasureCharacterBounds removes some glyphs when measuring a text using Style Script font. I am using the latest version of SixLabors.Fonts (2.0.3) with .NET 8 on Windows 10.

Steps to reproduce:

```
using SixLabors.Fonts;

var textContent = "Please enter the text";

var fontData = File.ReadAllBytes(@"<path to the font>\Style Script.ttf");

using var stream = new MemoryStream(fontData);

var fontsCollection = new FontCollection();
var fontFamily = fontsCollection.Add(stream);

var font = fontFamily.CreateFont(16f);

var renderOptions = new TextOptions(font)
{
    Dpi = 96,
    VerticalAlignment = VerticalAlignment.Top,
    HorizontalAlignment = HorizontalAlignment.Left,
    KerningMode = KerningMode.Auto
};

if (TextMeasurer.TryMeasureCharacterBounds(textContent.AsSpan(), renderOptions, out var bounds))
{
    Console.WriteLine($"Bounds length: {bounds.Length}");
    Console.WriteLine($"textContent length: {textContent.Length}");
}
else
{
    Console.WriteLine("Failed to measure bounds");
}

Console.ReadLine();
```
![image](https://github.com/SixLabors/Fonts/assets/38604580/92db07ca-2e7e-495f-99d1-1a1fa28bed1e)

After some debugging I found that LookupType4SubTable.TrySubstitution removes the second 'e' from "enter" and 't' from "the". This looks like a bug, though I am not sure.
Here's a ttf file of a font I am using: 
[Style Script.zip](https://github.com/user-attachments/files/15834980/Style.Script.zip)

**Comment by JimBobSquarePants** [comment, marked as answer]
There's no issue here. 

Here's what is happening:

- The glyphs at positions 10 and 11 `e (507) ` and `r (801)` are replaced by ligature `e_r (509)`. 
- Following that substitution, the glyphs at position 12 and `13 t (851)` `h (588)` are replaced by ligature `t_h (852)`

This leaves 19 glyphs in total instead of the original 21. 

As you can see the output is correct. 

![Please enter the text](https://github.com/SixLabors/Fonts/assets/385879/5ba0c2c3-faeb-4da6-bb4c-ea1e0f18f8cb)

If you wish to see a step by step demonstration of the process you can use this website. 
https://www.corvelsoftware.co.uk/crowbar/
